src/coreclr/jit/compiler.cpp

```diff
@@ -2019,6 +2019,201 @@ void Compiler::compInit(ArenaAllocator*       pAlloc,
+                    (RBM_FLT_CALLEE_TRASH & ~(RBM_DOUBLERET | RBM_F2 | RBM_F3 | RBM_F4 | RBM_F5 | RBM_F6 | RBM_F7)));
+    // The registers trashed by the CORINFO_HELP_INIT_PINVOKE_FRAME helper.
+    AllRegsMask_INIT_PINVOKE_FRAME_TRASH =
+        (AllRegsMask_CALLEE_TRASH | GprRegsMask(RBM_PINVOKE_TCB | RBM_PINVOKE_SCRATCH));
+
+    AllRegsMask_VALIDATE_INDIRECT_CALL_TRASH = GprRegsMask(RBM_INT_CALLEE_TRASH);
+
+#elif defined(TARGET_ARM64)
+
+    AllRegsMask_CALLEE_TRASH_NOGC    = GprRegsMask(RBM_CALLEE_TRASH_NOGC);
+    AllRegsMask_PROFILER_ENTER_TRASH = AllRegsMask((RBM_INT_CALLEE_TRASH & ~(RBM_ARG_REGS | RBM_ARG_RET_BUFF | RBM_FP)),
+                                                   (RBM_FLT_CALLEE_TRASH & ~RBM_FLTARG_REGS), RBM_MSK_CALLEE_TRASH);
+    // Registers killed by CORINFO_HELP_ASSIGN_REF and CORINFO_HELP_CHECKED_ASSIGN_REF.
+    AllRegsMask_CALLEE_TRASH_WRITEBARRIER = GprRegsMask(RBM_R14 | RBM_CALLEE_TRASH_NOGC);
+
+    // Registers no longer containing GC pointers after CORINFO_HELP_ASSIGN_REF and CORINFO_HELP_CHECKED_ASSIGN_REF.
+    AllRegsMask_CALLEE_GCTRASH_WRITEBARRIER = AllRegsMask_CALLEE_TRASH_NOGC;
+
+    // Registers killed by CORINFO_HELP_ASSIGN_BYREF.
+    AllRegsMask_CALLEE_TRASH_WRITEBARRIER_BYREF =
+        GprRegsMask(RBM_WRITE_BARRIER_DST_BYREF | RBM_WRITE_BARRIER_SRC_BYREF | RBM_CALLEE_TRASH_NOGC);
+
+    // Registers no longer containing GC pointers after CORINFO_HELP_ASSIGN_BYREF.
+    // Note that x13 and x14 are still valid byref pointers after this helper call, despite their value being changed.
+    AllRegsMask_CALLEE_GCTRASH_WRITEBARRIER_BYREF = AllRegsMask_CALLEE_TRASH_NOGC;
+
+    AllRegsMask_PROFILER_LEAVE_TRASH    = AllRegsMask_PROFILER_ENTER_TRASH;
+    AllRegsMask_PROFILER_TAILCALL_TRASH = AllRegsMask_PROFILER_ENTER_TRASH;
+
+    // The registers trashed by the CORINFO_HELP_STOP_FOR_GC helper
+    AllRegsMask_STOP_FOR_GC_TRASH = AllRegsMask_CALLEE_TRASH;
+    // The registers trashed by the CORINFO_HELP_INIT_PINVOKE_FRAME helper.
+    AllRegsMask_INIT_PINVOKE_FRAME_TRASH     = AllRegsMask_CALLEE_TRASH;
+    AllRegsMask_VALIDATE_INDIRECT_CALL_TRASH = GprRegsMask(RBM_VALIDATE_INDIRECT_CALL_TRASH);
+#endif
+
+#if defined(TARGET_ARM)
+    // profiler scratch remains gc live
+    AllRegsMask_PROF_FNC_LEAVE = AllRegsMask_PROFILER_LEAVE_TRASH & ~AllRegsMask_PROFILER_RET_SCRATCH;
+#else
+    AllRegsMask_PROF_FNC_LEAVE = AllRegsMask_PROFILER_LEAVE_TRASH;
+#endif // TARGET_ARM
+
+#ifdef TARGET_XARCH
+
+    // Make sure we copy the register info and initialize the
+    // trash regs after the underlying fields are initialized
+
+    const regMaskTP vtCalleeTrashRegs[TYP_COUNT]{
+#define DEF_TP(tn, nm, jitType, sz, sze, asze, st, al, regTyp, regFld, csr, ctr, tf) ctr,
+#include "typelist.h"
+#undef DEF_TP
+    };
+    memcpy(varTypeCalleeTrashRegs, vtCalleeTrashRegs, sizeof(regMaskTP) * TYP_COUNT);
+
+    if (codeGen != nullptr)
+    {
+        codeGen->CopyRegisterInfo();
+    }
+#endif // TARGET_XARCH
+}
```

Comment:
You may have said this earlier, but why can these not be static variables with values baked into the .dll?

Reply:
Because `compiler` object has ISA information that we use to determine the float/mask registers to include. https://github.com/dotnet/runtime/pull/98258#issuecomment-2048439472

Reply:
Do all of these sets need dynamic creation? Isn't it just the ones defined here that do?
https://github.com/dotnet/runtime/blob/b7c34460b56bbde2b5bc1e15b877ce475c7f22a6/src/coreclr/jit/codegeninterface.h#L62-L91